### PR TITLE
registry.k8s.io: Add a Cloudfront distribution

### PR DESCRIPTION
**Pull request by ameukam**
Related to:
  - https://github.com/kubernetes/registry.k8s.io/issues/194

Add a Cloudfront distribution that will serve blobs from the S3 buckets in GCP regions where there is no AWS presence.

### CODE REVIEW

**Comment by ameukam** [comment]
cc @upodroid @dims @BenTheElder

File: infra/aws/terraform/registry-k8s-io-prod/cloudfront.tf [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  }
+
+  origin {
+    domain_name = "prod-registry-k8s-io-eu-west-1.s3.eu-west-1.amazonaws.com"
+    origin_id   = "registry-k8s-io-prod-eu-west-1"
+
+    origin_shield {
+      enabled              = true
+      origin_shield_region = "eu-west-1"
+    }
```
Comment (xmudrii) [inline infra/aws/terraform/registry-k8s-io-prod/cloudfront.tf:57]: Why do we need to use origin shield in this case? My understanding is that this is useful when you have a non-AWS origin, but given that this is a S3 bucket, I don't think it provides many benefits while it adds to the cost.

File: infra/aws/terraform/registry-k8s-io-prod/locals.tf [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+locals {
+  assume_role_arn = "arn:aws:iam::${local.registry_k8s_io_prod_account_id}:role/OrganizationAccountAccessRole"
+
+  audit-account-name  = "k8s-infra-security-audit"
+  audit-account-index = index(data.aws_organizations_organization.current.accounts.*.name, local.audit-account-name)
+  audit-account-id    = data.aws_organizations_organization.current.accounts[local.audit-account-index].id
+
+  logging-account-name  = "k8s-infra-security-logs"
+  logging-account-index = index(data.aws_organizations_organization.current.accounts.*.name, local.logging-account-name)
+  logging-account-id    = data.aws_organizations_organization.current.accounts[local.audit-account-index].id
```
Comment (xmudrii) [inline infra/aws/terraform/registry-k8s-io-prod/locals.tf:26]: I _think_ this should be `logging-account-index` instead:
```suggestion
  logging-account-id    = data.aws_organizations_organization.current.accounts[local.logging-account-index].id
```

File: infra/aws/terraform/registry-k8s-io-prod/locals.tf [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+  networking-account-name  = "k8s-infra-networking"
+  networking-account-index = index(data.aws_organizations_organization.current.accounts.*.name, local.networking-account-name)
+  networking-account-id    = data.aws_organizations_organization.current.accounts[local.networking-account-index].id
+
+  registry_k8s_io_prod_name       = "k8s-infra-registry-k8s-io-prod"
```
Comment (xmudrii) [inline infra/aws/terraform/registry-k8s-io-prod/locals.tf:32]: Can we standardize on using `-` and `_`?

File: infra/aws/terraform/registry-k8s-io-prod/locals.tf [hunk 4]
```diff
@@ -0,0 +1,35 @@
+/*
+Copyright 2023 The Kubernetes Authors.
```
Comment (xmudrii) [inline infra/aws/terraform/registry-k8s-io-prod/locals.tf:2]: Should this be 2024 instead?

**Review by xmudrii — CHANGES_REQUESTED** [review]

File: infra/aws/terraform/registry-k8s-io-prod/locals.tf (hunk 4 above)
Comment (ameukam) [reply infra/aws/terraform/registry-k8s-io-prod/locals.tf:2]: Had this branch open in 2023.

File: infra/aws/terraform/registry-k8s-io-prod/locals.tf (hunk 3 above)
Comment (ameukam) [reply infra/aws/terraform/registry-k8s-io-prod/locals.tf]: fixed

File: infra/aws/terraform/registry-k8s-io-prod/cloudfront.tf [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+        // an Origin is present, and varies its response based on that. If we
+        // don't forward the header CloudFront is going to cache the first CORS
+        // response it receives, even if it's empty.
+        "Origin",
+      ]
+      query_string = false
+      cookies {
+        forward = "none"
+      }
+    }
```
Comment (xmudrii) [inline infra/aws/terraform/registry-k8s-io-prod/cloudfront.tf:79]: As per [Terraform docs](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cloudfront_distribution#default_cache_behavior), this option is deprecated and `cache_policy_id` is preferred instead

**Review by xmudrii — CHANGES_REQUESTED** [review]

File: infra/aws/terraform/registry-k8s-io-prod/cloudfront.tf (hunk 1 above)
Comment (xmudrii) [reply infra/aws/terraform/registry-k8s-io-prod/cloudfront.tf:57]: Apparently this is not a significant cost so we can leave it enabled

File: infra/aws/terraform/registry-k8s-io-prod/cloudfront.tf (hunk 5 above)
Comment (ameukam) [reply infra/aws/terraform/registry-k8s-io-prod/cloudfront.tf:79]: @xmudrii Cache policy added

**Review by xmudrii — APPROVED** [review]
/lgtm
/approve

/hold
for other folks to take a look

**Review by upodroid — APPROVED** [review]
LGTM

**Pull request merged**